pkg/controller/servicebindingrequest/servicebinder.go

```diff
@@ -236,6 +241,64 @@ func (b *serviceBinder) onError(
 	return requeueOnNotFound(err, requeueAfter)
 }
 
+// isApplicationSelectorEmpty returns true if applicationSelector is not declared in
+// the Service Binding Request.
+func isApplicationSelectorEmpty(
+	application v1alpha1.ApplicationSelector,
+) bool {
+	var emptyApplication v1alpha1.ApplicationSelector
+	if application == emptyApplication ||
+		application.ResourceRef == "" &&
+			application.LabelSelector.MatchLabels == nil {
+		return true
+	}
+	return false
```

Comment:
Perhaps this can be simplified:

```suggestion
	// in the case application.LabelSelector is a pointer:
	// return application.ResourceRef == "" && application.LabelSelector != nil && application.LabelSelector.MatchLabels == nil
	return application.ResourceRef == "" && application.LabelSelector.MatchLabels == nil
```

Reply:
Not committing this directly.
Made some changes here 7f886cd